### PR TITLE
[KAT-509] Manual control of CI.

### PR DESCRIPTION
**Pull request by arthurp**

### CODE REVIEW

**Comment by arthurp** [comment]
* ~Does not include documentation yet.~
* We could discuss whether the intermediate jobs are useful. They reduce code duplication and make it harder to make mistakes in the YAML, but they are also a bit of a mess themselves.

File: .github/workflows/python.yaml [hunk 1]
```diff
@@ -15,8 +15,69 @@ jobs:
 #      if: ${{ github.event_name == 'pull_request' }}
 #      with:
 #        access_token: ${{ github.token }}
+  do_python:
+    if: |
+      github.event_name != 'pull_request' ||
+      (
+        !contains(github.event.pull_request.body, '[no Python]')
```
Comment (ddn0) [inline .github/workflows/python.yaml:22]: I don't know if it is better or worse, but we can also trigger this on comments: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#discussion_comment

Comment (arthurp) [reply .github/workflows/python.yaml:22]: We could. We would have to trigger all our workflows on every comment an then make sure the jobs don't duplicate too much work, or everything is skipped in most cases. I'm not sure if github would leave previous successes in place if a later workflow run skips the previously succeeding job.

This seems tricky, and I wanted to keep this basic.

File: .github/workflows/cpp.yaml [hunk 2]
```diff
@@ -7,11 +7,72 @@ on:
   pull_request:
 
 jobs:
+  do_cpp:
+    if: |
+      github.event_name != 'pull_request' ||
+      (
+        !contains(github.event.pull_request.body, '[no C++]')
```
Comment (gliga) [inline .github/workflows/cpp.yaml:14]: I would prefer all lower case, but fine by me if we have to following lower+upper.  Is this case sensitive; can you make contains to be case insensitive

Comment (arthurp) [reply .github/workflows/cpp.yaml:14]: It's case insensitive. :-) I'll make sure that fact is in the docs.

**Comment by gliga** [comment]
It is a great change!

**Comment by arthurp** [comment]
I think this is ready for review. Please approve if you think it's in good shape.

**Review by ddn0 — APPROVED** [review]

**Pull request merged**